airbyte-integrations/bases/base-normalization/normalization/transform_catalog/utils.py

```diff
@@ -16,76 +17,125 @@ def remove_jinja(command: str) -> str:
     return str(command).replace("{{ ", "").replace(" }}", "")
 
 
+# transform oneOf nested list ot plain types list
+def get_plain_list_from_one_of_array(one_of_definition) -> list:
+    one_of_property = one_of_definition[data_type.ONE_OF_VAR_NAME]
+    plain_datatypes_list = list()
+    for prop in one_of_property:
+        plain_datatypes_list.append(prop[data_type.REF_TYPE_VAR_NAME])
+    return plain_datatypes_list
+
+
 def is_string(property_type) -> bool:
-    return property_type == "string" or "string" in property_type
+    if data_type.ONE_OF_VAR_NAME in property_type and data_type.ONE_OF_VAR_NAME in property_type:
+        plain_datatypes_list = get_plain_list_from_one_of_array(property_type)
+        return plain_datatypes_list == data_type.STRING_TYPE or data_type.STRING_TYPE in plain_datatypes_list
```

Comment:
The [integration test](https://github.com/airbytehq/airbyte/actions/runs/3614412995/jobs/6090834357) seems to be complaining about this line
```
BUILD FAILED in 26s
[python] .venv/bin/python -m mypy normalization --config-file /actions-runner/_work/airbyte/airbyte/pyproject.toml
	 normalization/transform_catalog/utils.py: note: In function "is_string":
	 normalization/transform_catalog/utils.py:32:16: error: Non-overlapping equality
	 check (left operand type: "List[Any]", right operand type: "str") 
	 [comparison-overlap]
	             return plain_datatypes_list == data_type.STRING_TYPE or data_t...
	                    ^
```

Reply:
Hi @grishick 
I believe that integration tests on CI fail because It generates thousands of files, but I haven't committed autogenerated files yet to make the review process easier.  Obviously, I will have to make them pass before merging :) 

Just to make sure, do you want me to make it works for both old and new protocols for a while? I may do that, but it would take me lots of additional time.  The original ticket says: "This updated version of normalization will never see old-style catalogs, because it will be released after the platform protocol versioning migrations, and normalization is released as part of platform (i.e. it's impossible for someone to upgrade normalization without also upgrading platform).". 

So I didn't spend time for keep supporting old-style also.

Reply:
Yes, my reasoning for supporting both versions was just as a fallback in case something weird happens. However, I think that's not necessary, so I am OK with not supporting the older version. If somehow an older platform ends up loading new normalization image it'll fail, but that's not a reasonable scenario given how we manage normalization version right now

Reply:
it'll become more of a possibility, since normalization version is moving into destination_definitions.yaml. I don't think there's any safeguard against downgrading normalization to a pre-protocol-v1 version other than "PR reviewer is careful"

though maybe there should be such a safeguard

Reply:
The problem here is that there are also lots of other places where the old "type" were used, not only in case method. To find all cases I had to do dozens run in debug mode for different connectors and check what actually comes here. Some of the methods also appear to be called in very unpredictable ways. So adding backward compatibility will take me some time (can't estimate it yet), as for now I will not know if the received "type" is an object or still should be treated as simple object. Also will need to dive deeper in test sqls. As not my goal will be not only make them pass, but also to extend to support versioning.
By the way, locally all tests pass, maybe something is not passed on Ci as I haven't committed autogenerated files to make the review process easier.

Reply:
Note from grooming session: we don't need to keep support for the old protocol version in normalization. The only  breaking scenario is old platform + new normalization, which is not possible unless someone deliberately updates hard-coded normalization version w/o updating the platform.